dist/js/select2.js

```diff
@@ -1802,7 +1802,11 @@ S2.define('select2/selection/allowClear',[
       }
     }
 
-    this.$element.val(this.placeholder.id).trigger('change');
+    if (typeof this.placeholder !== 'undefined') {
+        this.$element.val(this.placeholder.id).trigger('change');
+    } else {
+        this.$element.val('').trigger('change');
+    }
```

Comment:
We don't want it to create conflicts with other PRs, it'll be updated later. Looking forward to review the test case :guitar: